src/Battlescape/ActionMenuState.cpp

```diff
@@ -180,6 +180,56 @@ ActionMenuState::ActionMenuState(BattleAction *action, int x, int y) : _action(a
 	{
 		addItem(BA_USE, "STR_USE_MIND_PROBE", &id, Options::keyBattleActionItem1);
 	}
+	else if (weapon->getBattleType() == BT_SCRIPTED)
+	{
+		std::vector<SDLKey> hotKeys = {Options::keyBattleActionItem1, Options::keyBattleActionItem2, Options::keyBattleActionItem3, Options::keyBattleActionItem4, Options::keyBattleActionItem5};
+
+		if (!hotKeys.empty() && weapon->getCostUse().Time > 0)
+		{
+			addItem(BA_USE, "STR_USE_SCRIPTED", &id, hotKeys.back());
+			hotKeys.pop_back();
+		}
+		if (!hotKeys.empty() && weapon->getCostSnap().Time > 0)
+		{
+			addItem(BA_SNAPSHOT, "STR_SNAP_SCRIPTED", &id, hotKeys.back());
```

Comment:
why are the default action names different? shouldn't the override be done per weapon?

Reply:
That was to ensure, that the action names wouldn't be confused with the default ones. Since the scripted actions can be completely different.